packages/dds/tree/src/test/forestProxy.spec.ts

```diff
@@ -0,0 +1,73 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { strict as assert } from "assert";
+import { StoredSchemaRepository } from "../schema-stored";
+import { initializeForest, IEditableForest } from "../forest";
+import { JsonableTree, buildForest, proxifyForest } from "..";
+import { brand } from "../util";
+import { defaultSchemaPolicy } from "../feature-libraries";
+
+describe("forest-proxy", () => {
+	let forest: IEditableForest;
+	let proxy: any;
+
+	const content: JsonableTree = {
+		type: brand("Test:Person-1.0.0"),
+		fields: {
+			name: [{ value: "Adam", type: brand("String") }],
+			age: [{ value: 35, type: brand("Int32") }],
+			salary: [{ value: 10420.2, type: brand("Float32") }],
+			friends: [{ value: {
+				Mat: "Mat",
+			}, type: brand("Map<String>") }],
+			address: [{
+				fields: {
+					street: [{ value: "treeStreet", type: brand("String") }],
+					zip: [{ type: brand("String") }],
+				},
+				type: brand("Test:Address-1.0.0"),
+			}],
+		},
+	};
+
+	beforeEach(() => {
```

Comment:
I don't really like the mutable values outside of the test, setup in beforeEach pattern.

I think a more functional style, where we have a "buildTestProxy" function each test calls would be better.

This way it would be easier to add more tests that do different things (ex: allowing the test to select the content for the proxy could be done with an optional parameter) and the data flow is easier to follow.

Reply:
Done with the function. I intentionally left a test suite structure as it is (i.e. without `testCases` and a corresponding massive processing of those) since currently it tests more the way how the proxy is used rather than the way how the data types are handled as basically we have only two cases: a primitive and a node. It can be adjusted very easily if more complex cases/types will follow.